pos-print/src/main/java/com/clouway/pos/print/adapter/http/CashRegistersService.kt

```diff
@@ -0,0 +1,51 @@
+package com.clouway.pos.print.adapter.http
+
+import com.clouway.pos.print.core.CashRegister
+import com.clouway.pos.print.persistent.RegistersInMemoryRepository
+import com.google.common.collect.Lists
+import com.google.inject.Inject
+import com.google.sitebricks.At
+import com.google.sitebricks.client.transport.Json
+import com.google.sitebricks.headless.Reply
+import com.google.sitebricks.headless.Request
+import com.google.sitebricks.headless.Service
+import com.google.sitebricks.http.Get
+import com.google.sitebricks.http.Post
+import javax.servlet.http.HttpServletResponse.SC_CREATED
+
+/**
+ *@author Borislav Gadjev <borislav.gadjev@clouway.com>
+ */
+@Service
+@At("/v1/devices")
+class CashRegistersService @Inject constructor(private var repository: RegistersInMemoryRepository) {
+
+  @Get
+  fun getDevices(): Reply<*> {
+    val devicesDTO = adapt(repository.getAll())
+    return Reply.with(devicesDTO).`as`(Json::class.java).ok()
+  }
+
+  @Post
+  fun registerDevice(request: Request): Reply<*> {
+    val cashRegisterDTO = request.read(CashRegisterDTO::class.java).`as`(Json::class.java)
+
+    repository.register(adapt(cashRegisterDTO))
```

Comment:
You have to handle the case when device already exists with that ip address. 

I was mean 

```kotlin
repository.register(CashRegister(request.sourceIp, request.destination, request.description)) 
```
instead of extract this logic into an adapt function.